modules/ROOT/pages/spring-data-neo4j.adoc

```diff
@@ -1,318 +1,478 @@
-The direction property is _outgoing_ by default, so we must specify that the relationships are incoming on the `Movie` node.
+This class for person entities looks very similar to our `MovieEntity` class above.
+The `@Node` annotation defines that it is a database domain entity.
+A unique key field is identified (in this case, the `name` property), and a `born` property is defined as another attribute on this class.
+The constructor for the class follows the properties.
+
+Notice that we have not defined the relationships from a person back to a movie.
+In our use case, we only want to retrieve movies and the people involved in them.
+Our application does not need us to pull information for person entities separately, so we do not need to define the relationships back in the other direction.
+
+[NOTE]
+--
+If a domain needs to pull related entities on both sides, we would need to add the annotations and attributes from both sides.
+--
+
+[#define-repository]
+== Define a Spring Data repository
+
+Our repositories in the application will extend a repository provided out-of-the-box called the `ReactiveNeo4jRepository`.
+
+[NOTE]
+--
+If building an imperative application, you can extend the `Neo4jRepository`.
+Also, while technically not prohibited, it is not recommended or supported to mix imperative and reactive database access in the same application.
+--
 
-Finally, because these entities are all connected together, when we pull one entity in a request, it will pull the rest.
-When it pulls the other entities, it will follow the relationships back to the starting entity, which will go back to the related entities, creating a round-robin infinite recursion loop.
-To avoid this error, we can add an annotation to ignore certain fields when it traverses the relationships.
+Because our repositories are implementing reactive capabilities, we have access to the https://projectreactor.io/docs/core/release/reference/#mono[Mono^] and https://projectreactor.io/docs/core/release/reference/#flux[Flux^] reactive types from https://projectreactor.io/[Project Reactor^] for method returns.
+The `Mono` type returns 0 or 1 results, while the `Flux` returns 0 or n results.
+We would use a return type of `Mono` if we were expecting a single object back from the query and use a `Flux` type if we were expecting potentially multiple objects back from the query.
+
+[#movie-repository]
+=== Movie repository
 
-.Avoid Request Recursion
 [source,java]
 ----
-@NodeEntity
-public class Person {
-    ....
+public interface MovieRepository extends ReactiveNeo4jRepository<MovieEntity, String> {
+	Mono<MovieEntity> findOneByTitle(String title);
+}
+----
 
-    @JsonIgnoreProperties("person")
-    @Relationship(type = "ACTED_IN")
-    private List<Role> actedIn = new ArrayList<>();
+In this code block, we create a class type of `interface`, which is how Spring provides flexibility to interact with various data stores.
```

Comment:
```suggestion
In this code block, we create an `interface`, which is how Spring provides flexibility to interact with various data stores.
```
Or skip / merge this sentence with the following one because they both express about the same.

Reply:
further simplified in 2debe2837af6704d4e07e702cec53ba17f9da127